lib/util/result.ts

```diff
@@ -310,66 +309,68 @@ export class Result<T, E = Error> {
  *
  * All the methods resemble `Result` methods, but work asynchronously.
  */
-export class AsyncResult<T, E> extends Promise<Result<T, E>> {
-  constructor(
-    executor: (
-      resolve: (value: Result<T, E> | PromiseLike<Result<T, E>>) => void,
-      reject: (reason?: unknown) => void
-    ) => void
-  ) {
-    super(executor);
+export class AsyncResult<T, E> implements PromiseLike<Result<T, E>> {
+  private constructor(private asyncResult: Promise<Result<T, E>>) {}
+
+  then<TResult1 = Result<T, E>>(
+    onfulfilled?:
+      | ((value: Result<T, E>) => TResult1 | PromiseLike<TResult1>)
+      | undefined
+      | null
+  ): PromiseLike<TResult1> {
+    return this.asyncResult.then(onfulfilled);
   }
 
   static ok<T>(val: NonNullable<T>): AsyncResult<T, never> {
-    return new AsyncResult((resolve) => resolve(Result.ok(val)));
+    return new AsyncResult(Promise.resolve(Result.ok(val)));
   }
 
   static err<E>(err: NonNullable<E>): AsyncResult<never, E> {
-    return new AsyncResult((resolve) => resolve(Result.err(err)));
+    // eslint-disable-next-line promise/no-promise-in-callback
```

Comment:
🤔 why and what causes this 🤔

Reply:
I think it's bug of linter

Reply:
we should report it 😉

Reply:
If we do, I don't think it well be fixed anytime soon

Reply:
But I don't care too much about it

Reply:
i know, but a bug should be reported. we can leave this comment until upstream fix

Reply:
Let's merge it first, I'll see how to provide minimal reproduction

Reply:
https://github.com/eslint-community/eslint-plugin-promise/issues/445